src/charm.py

```diff
@@ -357,30 +383,46 @@ def on_update_status(self, _):
     def on_upgrade_charm(self, _):
         self.install_kubectl_plugin("kubectl-ko")
 
-    def patch_prometheus_resources(self, resources):
-        for res in resources:
+    def patch_prometheus_resources(self, resources, namespace, remove=False):
+        if remove:
             patch_file = self.render_template(
-                "patch-prometheus.yaml.j2", port=res["port"]
+                "patch-prometheus.yaml", scrape="null", port="null", remove=remove
             )
-            name = res["name"]
-            port = res["port"]
-            log.info(f"Rendered {name} patch w/ port: {port}, patching...")
+        for res in resources:
+            if not remove:
+                patch_file = self.render_template(
+                    "patch-prometheus.yaml",
+                    scrape="true",
+                    port=res["port"],
+                    remove=remove,
+                )
```

Comment:
i feel like that a tweak like this could simplify this logic:

```python
    def patch_prometheus_resources(self, resources, namespace, remove=False):
        for res in resources:
            patch_file = self.render_template(
                "patch-prometheus.yaml",
                scrape="null" if remove else True,
                port="null" if remove else res["port"],
                remove=remove,
            )
        ...
```